### DISCUSSION TITLE
Lines on edges of log of Fourier Amplitude figure

### DISCUSSION THREAD

**Discussion by hgrzy** [MPoL-dev/MPoL#90] [Q&A]
During last Thursday's discussion, we saw lines surrounding the outer edges of the log of the Fourier Amplitude (LFA) figure in the process of running an optimization loop. This occurred before reaching convergence. 

I'm now producing both the image plane figure and the LFA figure when running the loop to initialize the model with the dirty image. In the current version of the pull request for the HD143006 P2 tutorial, we go through 400 iterations during this process. Out of curiosity, I increased the number of iterations in the initialization of the model with the dirty image and am now producing the same lines that we saw in the optimization process (that included hyperparameters and a different loss function) even though the loss function is converged. Below includes: the model initialized with the dirty image in the image plane (at 950 iterations), the corresponding LFA (at 950 iterations), and the loss function vs iterations (note only every 50 iterations were logged). 

Why are we seeing different results with different loss functions even though they are both converging? Is it due to what data each loss function prefers?

Another thing to note, no matter how many iterations I use during this process, starting from step 0 to step 950, there is a resemblance of a cross hair dividing the LFA into four quadrants. Is that a property of the data?

![image](https://user-images.githubusercontent.com/35509654/126250144-1a280783-9bce-47f3-be78-1028fa7a64fc.png)
![image](https://user-images.githubusercontent.com/35509654/126250240-fbee8737-d134-4538-a968-e7cf6f53a203.png)

**Comment by hgrzy** [comment]
Could this be similar to the problem produced in optical astronomy with diffraction spikes? A cross hair feature seems to be prevalent in some of the optical images/fourier plane (though that may just be due to the cross hair on the instrument); my interpretation of the loop used to produce the above images is that as we are getting closer to the dirty image we are making the model's fourier plane closer to the visibilities in the data (which would be similar to adding more points of the aperture), which is why we get more lines on the edges- these would be comparable to the diffraction spikes in optical astronomy; some features I see in the LFA figure as I go through the iterations are apparent in this [paper](https://www.researchgate.net/publication/340097932_Kernel-phase_analysis_Aperture_modeling_prescriptions_that_minimize_calibration_errors/fulltext/5ea0adfba6fdcc88fc360c43/Kernel-phase-analysis-Aperture-modeling-prescriptions-that-minimize-calibration-errors.pdf?origin=publication_detail) for processing diffraction dominated images, namely figures 6 and 12- though it may be coincidental because I think one of the images is in the image plane. The figures contain a pattern of blobs surrounding the central feature, similarly to what I'm seeing in LFA figures:

<img width="317" alt="Screen Shot 2021-07-20 at 7 50 30 AM" src="https://user-images.githubusercontent.com/35509654/126319033-ca576e55-7538-4363-99f2-3441139fcaca.png">
<img width="317" alt="Screen Shot 2021-07-20 at 7 54 39 AM" src="https://user-images.githubusercontent.com/35509654/126319519-1c81f84f-9fa5-4ddc-90d2-55f68cf38f7f.png">

**Reply by hgrzy** [reply]
Update- looking into edge effects

**Reply by iancze** [reply]
I'm curious what the images look like for these Fourier representations? They do not seem to have any of the noise-dominated problems, even out to large q values.

**Reply by hgrzy** [reply]
These were the images I got corresponding to those LFAs:
![image](https://user-images.githubusercontent.com/35509654/126418593-6fdd64ea-f02f-4a20-89af-4a945c838c32.png)
![image](https://user-images.githubusercontent.com/35509654/126418610-0c6fb177-3ed2-4cb1-8654-e9426c6b64d6.png)

It's odd because they don't appear to be different from each other (or any other iteration for that matter). I dug a little more and found that sometimes the prominent lines in the LFA can be associated with edge effects so I'm wondering if more is to be seen in the image if I pan out. I'm also going to check the logs/logging function again tomorrow to double check that these are in fact the corresponding images since it's a bit suspicious.

**Reply by iancze** [reply]
Interesting. These seem to be quite similar to the dirty images (perhaps used for an initialization), so its possible something is not being updated correctly in the logs. What prior choices + regularizers are you using for these?

**Reply by hgrzy** [reply]
This loop is just from the initial optimization of the model with the dirty image

**Reply by hgrzy** [reply]
I was too curious and looked now haha. The logging function was incorrect. After correcting it, these were the results:

Iterations 50 - 950:

https://user-images.githubusercontent.com/35509654/126423803-603acbe9-c767-4aac-9738-eab590709f33.mov

Note the weird change in corners of the LFA between 5 and 6 seconds with no dramatic change in the image.

Then I increased the number of iterations as Robert did on thursday to attempt to get rid of the lines on the side.
Iterations 100-2950:

https://user-images.githubusercontent.com/35509654/126423958-278f77a4-8aa0-40d7-9f28-92b99b3a8ba7.mov

I don't think the final result of the model for this initialization looping will affect the popcorning, though the insight of why we see this behavior could be useful.

**Comment by iancze** [comment]
It's interesting that you are seeing this cross-hair response in the Fourier plane. 

The crosshair locations correspond to the u=0 and v=0 locations. Your comment about these corresponding to dominant features of the image is correct. What that means is that the image that corresponds to this LFA (nice [acronym](https://en.wikipedia.org/wiki/List_of_astronomy_acronyms) :) has more power at spatial frequencies (u=0, for all vs) and (all us, for v=0). I don't know if there are any good physical reasons to expect this behavior for a "natural" image, though. For analytic image/Fourier transform pairs, (e.g, a Gaussian ring and its FT), we wouldn't expect to see such features, so this makes me think it's an artifact of the optimization process.

As an example of how this comes about for an "unnatural" image, check out the FFT of the mock ALMA logo [here](https://en.wikipedia.org/wiki/List_of_astronomy_acronyms). The sharp features in the Fourier plane are from the strong features from the text and border outlines, many of which are preferentially in the x and y directions.

As a thought experiment, what would an image look like that *only* had power at  (u=0, for all vs) and (all us, for v=0)?

There is definitely a connection between diffraction spikes and the Fourier plane. For example, search on the term [Fourier Optics](https://en.wikipedia.org/wiki/Fourier_optics). I haven't yet wrapped my head around whether these specific features have a direct analogy to the diffraction spikes seen by [optical telescopes](https://en.wikipedia.org/wiki/Diffraction_spike), though, which are usually caused by occulting from support structures holding up the secondary mirrors. But at the end of the day, it all comes back to Fourier transforms so I'm sure any intuition gained there will be helpful in some respect.

**Reply by hgrzy** [reply]
Ah okay, being an artifact of the optimization process would be a good point to consider. Maybe I needed to do even more iterations to get the bars to go away then. 

Thank you for bringing up the example with the ALMA logo!

I would suspect the images to look like lines except rotated 90' from the orientation of their corresponding Fourier planes. 

Yeah, I think the direct comparison between features in optics and here that I made in a previous comment may actually be a coincidence. The main feature I was thinking of were the lines/spikes in the Fourier plane but after discussing where these come from in both cases I don't think they are direct analogies. Though I agree that there is a general connection between diffraction spikes and the Fourier plane! This might be why that "blob" pattern surrounding the central feature was prevalent in both cases.
